mm2src/mm2_db/src/indexed_db/drivers/cursor/cursor.rs

```diff
@@ -241,7 +241,7 @@ impl CursorDriver {
         })
     }
 
-    pub(crate) async fn next(&mut self) -> CursorResult<Option<(ItemId, Json)>> {
+    pub(crate) async fn next(&mut self, first_result_only: bool) -> CursorResult<Option<(ItemId, Json)>> {
```

Comment:
I believe this function should have consistent behavior without relying on conditional flags. Could we possibly create a new function called `fn first` and retain the existing function as is? The idea is to separate the common parts, which can then be called from both this function and `fn first`.

Reply:
did exactly this. thanks

Reply:
yeah, agree, now it looks beautiful.